File: dsa/3-arrays/challenges/02-palindrome/palindrome.py
import re


def is_a_palindrome(text):
    if not isinstance(text, str):
        raise ValueError("Input must be a string")
    if len(text) == 0:
        raise ValueError("String cannot be empty")
    cleaned_text = re.sub(r"[^a-zA-Z0-9]", "", text).lower()
    reversed_text = cleaned_text[::-1]
    print(reversed_text == cleaned_text)


is_a_palindrome("Menem !")

File: dsa/3-arrays/challenges/02-palindrome/palindrome.js
function isAPalindrome(text) {
    if (typeof text != "string") throw new Error('Input must be a string')
    if (text.length === 0) throw new Error('String cannot be empty')
    const cleanedText = text.replace(/[^a-zA-Z0-9]/g, '').toLowerCase()
    const reversed = cleanedText.split('').reverse().join('')
    console.log(reversed === cleanedText)
}

isAPalindrome("Menem !")